Description: Upstream changes introduced in version 2.1.0-1~maverickubuntu2
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 bumblebee (2.1.0-1~maverickubuntu2) maverick; urgency=high
 .
   * Based on git-changes v. 1.6.62..
 .
 The person named in the Author field signed this changelog entry.
Author: Martin Juhl (MrMEEE) <mj@casalogic.dk>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- bumblebee-2.1.0.orig/install-files/bumblebee-configuration
+++ bumblebee-2.1.0/install-files/bumblebee-configuration
@@ -271,6 +271,7 @@ while [ "$CONFIGMODE" != "MANUAL" ] && [
 done
 
 if [ "$CONFIGMODE" == "MANUAL" ]; then
+	POWERON="UNAVAILABLE"
 	if [ `${LSPCI} |grep VGA |wc -l` -eq 2 ]; then
 		NVIDIABUSID=`${LSPCI} | grep VGA | grep nVidia | awk '{ print "PCI:"$1 }' | sed s/\\\./:/g`
 	elif [ `${LSPCI} |grep 3D |wc -l` -eq 1 ]; then
@@ -287,20 +288,21 @@ fi
 sed -i "s/\([ \t]*BusID[ \t]*\)\".*\"/\1\"$NVIDIABUSID\"/g" /etc/X11/xorg.conf.nvidia
 
 if [ "$CONFIGMODE" == "MANUAL" ]; then
+	CONNECTEDMONITOR="UNKNOWN"
 	if [ -f /usr/local/bin/bumblebee-enablecard ]; then
 		/usr/local/bin/bumblebee-enablecard
 	fi
 	if [ -f /usr/bin/bumblebee-enablecard ]; then
 		/usr/bin/bumblebee-enablecard
 	fi
-
 	if [ "$DISTRO" == "UBUNTU" ] ; then
 		${MODPROBE} nvidia-current
-
-		if [ `LD_LIBRARY_PATH=/usr/lib/nvidia-current /usr/lib/nvidia-current/bin/nvidia-xconfig --query-gpu-info | grep "Display Devices" | cut -f2 -d":"` -gt 0 ] ; then
-			CONNECTEDMONITOR=`LD_LIBRARY_PATH=/usr/lib/nvidia-current /usr/lib/nvidia-current/bin/nvidia-xconfig --query-gpu-info | grep "Display Device 0" | cut -f2 -d\( | cut -f1 -d\)`
+		if [ "`lspci -v -s 01:00.0 |grep "Unknown header type"`" == "" ]; then
+			if [ `LD_LIBRARY_PATH=/usr/lib/nvidia-current /usr/lib/nvidia-current/bin/nvidia-xconfig --query-gpu-info | grep "Display Devices" | cut -f2 -d":"` -gt 0 ] ; then
+				CONNECTEDMONITOR=`LD_LIBRARY_PATH=/usr/lib/nvidia-current /usr/lib/nvidia-current/bin/nvidia-xconfig --query-gpu-info | grep "Display Device 0" | cut -f2 -d\( | cut -f1 -d\)`
+			fi
 		fi
-		${MODPROBE} -r nvidia-current
+		${MODPROBE} -r nvidia
 	fi
 	if [ "$DISTRO" == "OPENSUSE" ] ; then
 		${MODPROBE} nvidia 2>&1 >/dev/null
@@ -321,7 +323,6 @@ if [ "$CONFIGMODE" == "MANUAL" ]; then
 	if [ -f /usr/bin/bumblebee-disablecard ]; then
 		/usr/bin/bumblebee-disablecard
 	fi
-
 	if [ "$CONNECTEDMONITOR" != "CRT-0" ] && [ "$CONNECTEDMONITOR" != "DFP-0" ];then
 		LIST_SIZE=$(($LINES - 10))
 choice=$(whiptail --nocancel --menu \
@@ -343,151 +344,146 @@ fi
 
 sed -i "s/\([ \t]*Option[ \t]*\"ConnectedMonitor\"[ \t]*\)".*"/\1\"$CONNECTEDMONITOR\"/g" /etc/X11/xorg.conf.nvidia
 
-if [ $CONFIGMODE != "MANUAL" ] ; then
+#
+#	Always try to enable the card before switching scripts
+#
+if [ -f /usr/local/bin/bumblebee-enablecard ]; then
+	/usr/local/bin/bumblebee-enablecard
+fi
+if [ -f /usr/bin/bumblebee-enablecard ]; then
+	/usr/bin/bumblebee-enablecard
+fi
+if [ "$DISTRO" == "UBUNTU" ] ; then
+	if [ ! "$POWERON" == "UNAVAILABLE" ] ; then
+		echo $POWERON | xxd -r -p > /usr/local/bin/bumblebee-enablecard
+		echo $POWEROFF | xxd -r -p > /usr/local/bin/bumblebee-disablecard
+	else
+		whiptail --msgbox \
+" No power-on/off configuration has been reported for your machine,\n"\
+"please manually enter configuration in:\n\n"\
+"/usr/local/bin/bumblebee-enablecard and /usr/local/bin/bumblebee-disablecard\n\n"\
+"Some examples are available in /usr/share/doc/bumblebee/.\n\n"\
+"Press any key to continue..." $LINES $COLUMNS
+		if [ ! -f /usr/local/bin/bumblebee-enablecard ] ; then
+			# Not installed
+			cp /usr/share/doc/bumblebee/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
+		else
+			# Already Exists
+			echo
+			echo "nVidia card enable-script: /usr/local/bin/bumblebee-enablecard exists moving old script to /usr/local/bin/bumblebee-enablecard.old."
+			echo
+			mv /usr/local/bin/bumblebee-enablecard /usr/local/bin/bumblebee-enablecard.old
+			cp /usr/share/doc/bumblebee/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
+		fi
+		if [ ! -f /usr/local/bin/bumblebee-disablecard ]; then
+			# Not installed
+			cp /usr/share/doc/bumblebee/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
+		else
+			# Already Exists
+			echo
+			echo "nVidia card disable-script: /usr/local/bin/bumblebee-disablecard exists, moving old script to /usr/local/bin/bumblebee-disablecard.old."
+			echo
+			mv /usr/local/bin/bumblebee-disablecard /usr/local/bin/bumblebee-disablecard.old
+			cp /usr/share/doc/bumblebee/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
+		fi
+	fi
+fi
+if [ "$DISTRO" == "OPENSUSE" ] ; then
 	#
-	#	Always try to enable the card before switching scripts
+	#	Setup new scripts
 	#
-	if [ -f /usr/local/bin/bumblebee-enablecard ]; then
-		/usr/local/bin/bumblebee-enablecard
-	fi
-	if [ -f /usr/bin/bumblebee-enablecard ]; then
-		/usr/bin/bumblebee-enablecard
-	fi
-
-	if [ "$DISTRO" == "UBUNTU" ] ; then
-		if [ $POWERON != "UNAVAILABLE" ] ; then
+	if [ ! "$POWERON" == "UNAVAILABLE" ] ; then
+		if [ -f /usr/local/bin/optirun ] ; then
+			#
+			#	Script install
+			#
 			echo $POWERON | xxd -r -p > /usr/local/bin/bumblebee-enablecard
 			echo $POWEROFF | xxd -r -p > /usr/local/bin/bumblebee-disablecard
-		else
+		elif [ -f /usr/bin/optirun ] ; then
+			#
+			#	Repository install
+			#
+			echo $POWERON | xxd -r -p > /usr/bin/bumblebee-enablecard
+			echo $POWEROFF | xxd -r -p > /usr/bin/bumblebee-disablecard
+		fi
+	else
+		if [ -f /usr/local/bin/optirun ] ; then
+			#
+			#	Script install
+			#
 			whiptail --msgbox \
-" No power-on/off configuration has been reported for your machine,\n"\
+"No power-on/off configuration has been reported for your machine,\n"\
 "please manually enter configuration in:\n\n"\
 "/usr/local/bin/bumblebee-enablecard and /usr/local/bin/bumblebee-disablecard\n\n"\
-"Some examples are available in /usr/share/doc/bumblebee/.\n\n"\
+"Some examples are available in /usr/share/doc/packages/bumblebee/examples/.\n\n"\
 "Press any key to continue..." $LINES $COLUMNS
 			if [ ! -f /usr/local/bin/bumblebee-enablecard ] ; then
 				# Not installed
-				cp /usr/share/doc/bumblebee/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
 			else
 				# Already Exists
 				echo
 				echo "nVidia card enable-script: /usr/local/bin/bumblebee-enablecard exists moving old script to /usr/local/bin/bumblebee-enablecard.old."
 				echo
 				mv /usr/local/bin/bumblebee-enablecard /usr/local/bin/bumblebee-enablecard.old
-				cp /usr/share/doc/bumblebee/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
 			fi
-			if [ ! -f /usr/local/bin/bumblebee-disablecard ]; then
+			sed -i "s?INSERTPOWERON?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/local/bin/bumblebee-enablecard
+
+			if [ ! -f /usr/local/bin/bumblebee-disablecard ] ; then
 				# Not installed
-				cp /usr/share/doc/bumblebee/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
 			else
 				# Already Exists
 				echo
 				echo "nVidia card disable-script: /usr/local/bin/bumblebee-disablecard exists, moving old script to /usr/local/bin/bumblebee-disablecard.old."
 				echo
 				mv /usr/local/bin/bumblebee-disablecard /usr/local/bin/bumblebee-disablecard.old
-				cp /usr/share/doc/bumblebee/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
 			fi
-		fi
-	fi
-
-	if [ "$DISTRO" == "OPENSUSE" ] ; then
-		#
-		#	Setup new scripts
-		#
-		if [ $POWERON != "UNAVAILABLE" ] ; then
-			if [ -f /usr/local/bin/optirun ] ; then
-				#
-				#	Script install
-				#
-				echo $POWERON | xxd -r -p > /usr/local/bin/bumblebee-enablecard
-				echo $POWEROFF | xxd -r -p > /usr/local/bin/bumblebee-disablecard
-			elif [ -f /usr/bin/optirun ] ; then
-				#
-				#	Repository install
-				#
-				echo $POWERON | xxd -r -p > /usr/bin/bumblebee-enablecard
-				echo $POWEROFF | xxd -r -p > /usr/bin/bumblebee-disablecard
-			fi
-		else
-			if [ -f /usr/local/bin/optirun ] ; then
-				#
-				#	Script install
-				#
-				whiptail --msgbox \
-"No power-on/off configuration has been reported for your machine,\n"\
-"please manually enter configuration in:\n\n"\
-"/usr/local/bin/bumblebee-enablecard and /usr/local/bin/bumblebee-disablecard\n\n"\
-"Some examples are available in /usr/share/doc/packages/bumblebee/examples/.\n\n"\
-"Press any key to continue..." $LINES $COLUMNS
-				if [ ! -f /usr/local/bin/bumblebee-enablecard ] ; then
-					# Not installed
-					cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
-				else
-					# Already Exists
-					echo
-					echo "nVidia card enable-script: /usr/local/bin/bumblebee-enablecard exists moving old script to /usr/local/bin/bumblebee-enablecard.old."
-					echo
-					mv /usr/local/bin/bumblebee-enablecard /usr/local/bin/bumblebee-enablecard.old
-					cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/local/bin/bumblebee-enablecard
-				fi
-				sed -i "s?INSERTPOWERON?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/local/bin/bumblebee-enablecard
-
-				if [ ! -f /usr/local/bin/bumblebee-disablecard ] ; then
-					# Not installed
-					cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
-				else
-					# Already Exists
-					echo
-					echo "nVidia card disable-script: /usr/local/bin/bumblebee-disablecard exists, moving old script to /usr/local/bin/bumblebee-disablecard.old."
-					echo
-					mv /usr/local/bin/bumblebee-disablecard /usr/local/bin/bumblebee-disablecard.old
-					cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/local/bin/bumblebee-disablecard
-				fi
-				sed -i "s?INSERTPOWEROFF?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/local/bin/bumblebee-disablecard
-
-				chmod +x /usr/local/bin/bumblebee*
-			elif [ -f /usr/bin/optirun ] ; then
-				#
-				#	Repository install
-				#
-				whiptail --msgbox \
+			sed -i "s?INSERTPOWEROFF?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/local/bin/bumblebee-disablecard
+			chmod +x /usr/local/bin/bumblebee*
+		elif [ -f /usr/bin/optirun ] ; then
+			#
+			#	Repository install
+			#
+			whiptail --msgbox \
 "No power-on/off configuration has been reported for your machine,\n"\
 "please manually enter configuration in:\n\n"\
 "/usr/bin/bumblebee-enablecard and /usr/bin/bumblebee-disablecard\n\n"\
 "Some examples are available in /usr/share/doc/packages/bumblebee/examples/.\n\n"\
 "Press any key to continue..." $LINES $COLUMNS
-				if [ ! -f /usr/bin/bumblebee-enablecard ] ; then
-					# Not installed
-					cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/bin/bumblebee-enablecard
-				else
-					# Already Exists
-					echo
-					echo "nVidia card enable-script: /usr/bin/bumblebee-enablecard exists moving old script to /usr/bin/bumblebee-enablecard.old."
-					echo
-					mv /usr/bin/bumblebee-enablecard /usr/bin/bumblebee-enablecard.old
-					cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/bin/bumblebee-enablecard
-				fi
-				sed -i "s?INSERTPOWERON?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/bin/bumblebee-enablecard
-
-				if [ ! -f /usr/bin/bumblebee-disablecard ] ; then
-					# Not installed
-					cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/bin/bumblebee-disablecard
-				else
-					# Already Exists
-					echo
-					echo "nVidia card disable-script: /usr/bin/bumblebee-disablecard exists, moving old script to /usr/bin/bumblebee-disablecard.old."
-					echo
-					mv /usr/bin/bumblebee-disablecard /usr/bin/bumblebee-disablecard.old
-					cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/bin/bumblebee-disablecard
-				fi
-				sed -i "s?INSERTPOWEROFF?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/bin/bumblebee-disablecard
+			if [ ! -f /usr/bin/bumblebee-enablecard ] ; then
+				# Not installed
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/bin/bumblebee-enablecard
+			else
+				# Already Exists
+				echo
+				echo "nVidia card enable-script: /usr/bin/bumblebee-enablecard exists moving old script to /usr/bin/bumblebee-enablecard.old."
+				echo
+				mv /usr/bin/bumblebee-enablecard /usr/bin/bumblebee-enablecard.old
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-enablecard.template /usr/bin/bumblebee-enablecard
+			fi
+			sed -i "s?INSERTPOWERON?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/bin/bumblebee-enablecard
 
-				chmod +x /usr/bin/bumblebee*
+			if [ ! -f /usr/bin/bumblebee-disablecard ] ; then
+				# Not installed
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/bin/bumblebee-disablecard
+			else
+				# Already Exists
+				echo
+				echo "nVidia card disable-script: /usr/bin/bumblebee-disablecard exists, moving old script to /usr/bin/bumblebee-disablecard.old."
+				echo
+				mv /usr/bin/bumblebee-disablecard /usr/bin/bumblebee-disablecard.old
+				cp /usr/share/doc/packages/bumblebee/examples/bumblebee-disablecard.template /usr/bin/bumblebee-disablecard
 			fi
+			sed -i "s?INSERTPOWEROFF?echo \"Please run bumblebee-configuration to enable power management.\"?g" /usr/bin/bumblebee-disablecard
+			chmod +x /usr/bin/bumblebee*
 		fi
 	fi
 fi
 
+
 if [ "$DISTRO" == "UBUNTU" ] ; then
 	chmod +x /usr/local/bin/bumblebee*
 	update-rc.d bumblebee defaults
